graylog2-server/src/main/java/org/graylog2/cluster/NodeService.java

```diff
@@ -31,10 +31,10 @@
 @Deprecated(since = "6.0")
 public interface NodeService {
 
-    boolean registerServer(String nodeId, boolean isLeader, URI httpPublishUri, String clusterUri, String hostname, boolean isProcessing, LoadBalancerStatus lbStatus);
+    boolean registerServer(String nodeId, boolean isLeader, URI httpPublishUri, String clusterUri, String hostname, boolean isProcessing, Lifecycle lifecycle);
 
-    default boolean registerServer(String nodeId, boolean isLeader, URI httpPublishUri, String hostname, boolean isProcessing, LoadBalancerStatus lbStatus) {
-        return registerServer(nodeId, isLeader, httpPublishUri, null, hostname, isProcessing, lbStatus);
+    default boolean registerServer(String nodeId, boolean isLeader, URI httpPublishUri, String hostname, boolean isProcessing, Lifecycle lifecycle) {
```

Comment:
did we really need these changes in the deprecated interface/implementation/test classes? looks like `registerServer` is only used in test classes anyway and the Lifecycle/Load Balancer status is ignored there.

Reply:
I think we do, as long as we are using the deprecated interface in our own code, like in the NodeTitleDecorator. If we refactor this to the new typed node service, we can remove all the usages.

Without adapting the signature of these methods, they will build invalid DTOs and fail immediately.

Reply:
ok, it is cleaner anyhow to have these working as they should even if they are deprecated